addons/account/report/account_aged_partner_balance.py

```diff
@@ -151,15 +151,23 @@ def _get_lines(self, form):
         for i in range(5):
             args_list = (tuple(move_state), tuple(self.ACCOUNT_TYPE), tuple(partner_ids),self.date_from,)
             dates_query = '(COALESCE(l.date_maturity,l.date)'
+            date_partial = ''
+            arg_partial = ()
             if form[str(i)]['start'] and form[str(i)]['stop']:
                 dates_query += ' BETWEEN %s AND %s)'
                 args_list += (form[str(i)]['start'], form[str(i)]['stop'])
+                date_partial = 'AND l.date <= %s'
```

Comment:
I don't understand why you don't use BETWEEN in this case.
1. If you have start and stop date => you add stop date in the condition
2. If you have start date only => you add start date in the condition
3. If you have stop date => you add stop date in the condition

Why case 1 is not a combination of cases 2 and 3?